edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -86,9 +91,40 @@ internal static CreateContainerParameters GetCreateOptions(string createOptions)
         }
 
         internal static string ValidateAndGetImage(string image)
+        {
+            return ValidateAndGetImage(image, new EnvironmentWrapper());
+        }
+
+        internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env)
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
+
+            if (image[0] == '$')
+            {
+                Match matchHost = ImageUpstreamRegex.Match(image);
+                if (matchHost.Success
+                    && (matchHost.Groups["path"]?.Length > 0)
+                    && (env != null))
+                {
+                    string hostAddress = env.GetVariable(Core.Constants.GatewayHostnameVariableName);
+
+                    if (hostAddress != null)
+                    {
+                        image = hostAddress + matchHost.Groups["path"].Value;
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Could not find environment variable: {Core.Constants.GatewayHostnameVariableName}");
+                    }
+                }
+                else
+                {
+                    throw new ArgumentException($"Image {image} is not in the right format. If your intention is to use an environment variable, check the port is specified.");
```

Comment:
This should not be argument exception, which means invalid input from caller.  This should be more appropriate to throw InvalidOperationException.  Please confirm with @varunpuranik